File: buildit-pc-backend/app/Models/UserBuild.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;


class UserBuild extends Model
{
    use HasFactory;
    protected $table = "user_build";
    protected $primaryKey = "id";

    public $timestamps = true; // set to true l8r
    
    protected $fillable = [
        'cpu', 
        'gpu', 
        'motherboard', 
        'ram', 
        'storage', 
        'psu', 
        'user_id', 
        'name', 
        'description', 
        'total_tdp', 
        'total_price', 
        'benchmarkScore',
        'image_url'
    ];

    
    public function cpu()
    {
        return $this->belongsTo(CPU::class, 'cpu')->select('id', 'brand', 'name', 'model'); 
    }

    public function gpu()
    {
        return $this->belongsTo(GPU::class, 'gpu')->select('id', 'brand', 'model');
    }

    public function motherboard()
    {
        return $this->belongsTo(Motherboard::class, 'motherboard')->select('id', 'brand', 'model');
    }

    public function ram()
    {
        return $this->belongsTo(RAM::class, 'ram')->select('id', 'model');
    }

    public function storage()
    {
        return $this->belongsTo(Storage::class, 'storage')->select('id', 'brand', 'model');
    }

    public function psu()
    {
        return $this->belongsTo(PSU::class, 'psu')->select('id', 'brand', 'model');
    }


}

File: frontend/lib/services/userBuild.dart
import 'dart:io';

class UserBuild {
  final int? id;
  final String name;
  final String? description;
  final int cpu;
  final int gpu;
  final int motherboard;
  final int ram;
  final int storage;
  final int psu;
  final int? totalTdp;
  final double? totalPrice;
  final int? benchmarkScore;
  final int? user_id;
  File? imageFile;
  final String? image_url;

  UserBuild({
    required this.id,
    required this.name,
    required this.description,
    required this.cpu,
    required this.gpu,
    required this.motherboard,
    required this.ram,
    required this.storage,
    required this.psu,
    required this.totalTdp,
    required this.totalPrice,
    this.benchmarkScore,
    required this.user_id,
    this.imageFile,
    this.image_url,
  });

  factory UserBuild.fromJson(Map<String, dynamic> json) {
    return UserBuild(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      cpu: json['cpu'],
      gpu: json['gpu'],
      motherboard: json['motherboard'],
      ram: json['ram'],
      storage: json['storage'],
      psu: json['psu'],
      totalTdp: json['total_tdp'],
      totalPrice: json['total_price'].toDouble(),
      benchmarkScore: json['benchmarkScore'],
      user_id: json['user_id'],
      image_url : json['image_url'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'cpu': cpu,
      'gpu': gpu,
      'motherboard': motherboard,
      'ram': ram,
      'storage': storage,
      'psu': psu,
      'total_tdp': totalTdp,
      'total_price': totalPrice,
      'benchmarkScore': benchmarkScore,
      'user_id': user_id,
      'image_url' : image_url,
    };
  }
}
